projects/hslayers/src/components/draw/draw.service.ts

```diff
@@ -612,6 +640,50 @@ export class HsDrawService {
         },
         this
       );
+
+      //Add snap interaction -  must be added after the Modify and Draw interactions
+      const snapSourceToBeUsed = this.snapSource
+        ? this.snapSource
+        : this.source;
+      this.toggleSnapping(snapSourceToBeUsed);
     });
   }
+
+  /**
+   * Handle snap interaction changes
+   * Remove snap interaction if it already exists, recreate it if source is provided.
+   */
+  toggleSnapping(source?: VectorSource): void {
+    this.HsMapService.loaded().then((map) => {
+      this.snapSource = source ? source : this.snapSource;
+      if (this.snapActive && this.snapSource) {
+        if (this.snap) {
+          map.removeInteraction(this.snap);
+        }
+        this.snap = new Snap({
+          source: this.snapSource,
+        });
+        map.addInteraction(this.snap);
+      } else {
+        if (this.snap) {
+          map.removeInteraction(this.snap);
+          // this.snapLayer = null;
+        }
+      }
+    });
+  }
+  /**
+   * Changes layer source of snap interaction
+   */
+  changeSnapSource(layer: Layer): void {
+    const isLayerClustered = this.HsLayerUtilsService.isLayerClustered(layer);
+    let snapSourceToBeUsed;
```

Comment:
Ternary operator?

Reply:
Sure, itll be cleaner. I just coppied this part from else where so I might as well refactor both occurences.

Reply:
great, thanks